simtools/model/mirrors.py

```diff
@@ -60,27 +61,55 @@ def _read_mirror_list_from_ecsv(self):
 
         self.mirror_table = Table.read(self._mirror_list_file, format="ascii.ecsv")
         self._logger.debug(f"Reading mirror properties from {self._mirror_list_file}")
-        try:
-            self.shape_type = u.Quantity(self.mirror_table["shape_type"])[0].value
-            self.mirror_diameter = u.Quantity(self.mirror_table["mirror_diameter"])[0].value
-            self.number_of_mirrors = len(self.mirror_table["focal_length"])
 
+        self.number_of_mirrors = np.shape(self.mirror_table)[0]
+        self._logger.debug(f"Number of Mirrors = {self.number_of_mirrors}")
+
+        if self.number_of_mirrors == 0:
+            msg = "Problem reading mirror list file"
+            self._logger.error(msg)
+            raise InvalidMirrorListFile
+
+        try:
+            self.shape_type = u.Quantity(self.mirror_table["shape_type"])[0]
             self._logger.debug(f"Mirror shape_type = {self.shape_type}")
-            self._logger.debug(f"Mirror diameter = {self.mirror_diameter}")
-            self._logger.debug(f"Number of Mirrors = {self.number_of_mirrors}")
         except KeyError:
             self._logger.debug("Mirror shape_type not in mirror file")
-            self.number_of_mirrors = len(self.mirror_table["mirror_id"])
+            try:
+                self.shape_type = self.parameters["mirror_panel_shape"]["Value"]
+            except TypeError:
+                self._logger.debug("Mirror mirror_panel_shape not contained in db")
 
-        if "focal_length" not in self.mirror_table.colnames:
-            self.mirror_table["focal_length"] = (
-                self.mirror_table["mirror_diameter"].to("cm").value / 4.0
-            )
+        try:
+            self.mirror_diameter = u.Quantity(self.mirror_table["mirror_diameter"])[0]
+            self._logger.debug(f"Mirror diameter = {self.mirror_diameter}")
+        except KeyError:
+            self._logger.debug("Mirror mirror_panel_diameter not in mirror file")
+            try:
```

Comment:
A general comment about all of these nested `try..except`. I am not sure if they should be nested or not. I remember Gernot had an issue that the second except was never reached. Please make sure to include tests that reach all cases to make sure it works. If it doesn't, try to put the second `try..except` at the same level as the first.